iotdb-core/datanode/src/main/java/org/apache/iotdb/db/queryengine/execution/operator/source/relational/aggregation/grouped/GroupedCountIfAccumulator.java

```diff
@@ -0,0 +1,91 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing,
+ * software distributed under the License is distributed on an
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+ * KIND, either express or implied.  See the License for the
+ * specific language governing permissions and limitations
+ * under the License.
+ */
+
+package org.apache.iotdb.db.queryengine.execution.operator.source.relational.aggregation.grouped;
+
+import org.apache.iotdb.db.queryengine.execution.operator.source.relational.aggregation.grouped.array.LongBigArray;
+
+import org.apache.tsfile.block.column.Column;
+import org.apache.tsfile.block.column.ColumnBuilder;
+import org.apache.tsfile.enums.TSDataType;
+import org.apache.tsfile.utils.RamUsageEstimator;
+
+import java.util.List;
+
+import static com.google.common.base.Preconditions.checkArgument;
+import static org.apache.tsfile.enums.TSDataType.BOOLEAN;
+
+public class GroupedCountIfAccumulator implements GroupedAccumulator {
+  private final LongBigArray countValues = new LongBigArray(0L);
+
+  private static final long INSTANCE_SIZE =
+      RamUsageEstimator.shallowSizeOfInstance(GroupedCountIfAccumulator.class);
+
+  public GroupedCountIfAccumulator(List<TSDataType> seriesDataType) {
+    checkArgument(
+        seriesDataType.size() == 1 && seriesDataType.get(0) == BOOLEAN,
+        "argument of count_if should be one boolean expression");
+  }
+
+  @Override
+  public long getEstimatedSize() {
+    return INSTANCE_SIZE + countValues.sizeOf();
+  }
+
+  @Override
+  public void setGroupCount(long groupCount) {
+    countValues.ensureCapacity(groupCount);
+  }
+
+  @Override
+  public void addInput(int[] groupIds, Column[] arguments) {
+    for (int i = 0; i < groupIds.length; i++) {
+      if (arguments[0].getBoolean(i)) {
```

Comment:
firstly should check the nullability of arguments by if (arguments[0].isNull(i))